test/test_view/test_assign_task_worker.py

```diff
@@ -0,0 +1,88 @@
+        task_user_pref = dict(assign_user=[user.email_addr])
+        task = TaskFactory.create(project=project, user_pref=task_user_pref)
+
+        url = '/project/%s/tasks/assign-workersupdate?api_key=%s' % (project.short_name, project.owner.api_key)
+        req_data = dict(taskId=str(task.id))
+        res = self.app.post(url, content_type='application/json',
+                            data=json.dumps(req_data))
+        res_data = json.loads(res.data)
+        assert len(res_data['assign_users']) == 1, res_data['assign_users']
+        assert res_data['assign_users'][0]['email'] == user.email_addr
+        assert res_data['assign_users'][0]['fullname'] == user.fullname
+
+        all_user_emails = [u['email'] for u in res_data['all_users']]
+        assert user.email_addr not in all_user_emails, "existing users should be excluded from user list"
+
+    @with_context
+    def test_get_users_single_task_2(self):
+        """Test a single task with assign_user invalid email addr."""
+        project = ProjectFactory.create(published=True)
+        user = UserFactory.create(email_addr='a@a.com', fullname="test_user")
+        invalid_email_addr = "invalid@email"
+        task_user_pref = dict(assign_user=[user.email_addr, invalid_email_addr])
+        task = TaskFactory.create(project=project, user_pref=task_user_pref)
+
+        url = '/project/%s/tasks/assign-workersupdate?api_key=%s' % (project.short_name, project.owner.api_key)
+        req_data = dict(taskId=str(task.id))
+        res = self.app.post(url, content_type='application/json',
+                            data=json.dumps(req_data))
+        res_data = json.loads(res.data)
+        assert len(res_data['assign_users']) == 2, res_data['assign_users']
+        assert res_data['assign_users'][0]['email'] == invalid_email_addr
+        assert res_data['assign_users'][0]['fullname'] == invalid_email_addr + " (user not found)"
+
+        all_user_emails = [u['email'] for u in res_data['all_users']]
+        assert invalid_email_addr not in all_user_emails, "existing users should be excluded from user list"
+
+    @with_context
+    def test_get_users_bulk_task_0(self):
+        """Test a bulk task without assign_user."""
+        project = ProjectFactory.create(published=True)
+        user = UserFactory.create(email_addr='a@a.com', fullname="test_user")
+
+        task1_user_pref = dict(assign_user=[user.email_addr])
+
+        task1 = TaskFactory.create(project=project,  user_pref=task1_user_pref)
+        task2 = TaskFactory.create(project=project,  user_pref=task1_user_pref)
+        task_repo.update(task1)
+        task_repo.update(task2)
+
+        url = '/project/%s/tasks/assign-workersupdate?api_key=%s' % (project.short_name, project.owner.api_key)
+        req_data = dict(taskId=None)
+        res = self.app.post(url, content_type='application/json',
+                            data=json.dumps(req_data))
```

Comment:
new line can be added at the end here

Reply:
is this test expected to test/validate return value or expected functions called with desired params?